### PR TITLE
Selection Sort Algorithm

### PR DESCRIPTION
**Pull request by Miraddo**
Added selection sort algorithm with test and benchmark with the `iterative` process.

### CODE REVIEW

File: topics/go/algorithms/sorting/selection/selection.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+// selectionSortIterative takes a random list of numbers and uses the
+// `iterative` process to sort it and return the sorted list.
+func selectionSortIterative(randomList []int) []int {
+
+	// Loop until the random list will be sorted.
+	for i := range randomList {
+		index := i
+
+		// Continue the loop until to found the smallest number in the list.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/sorting/selection/selection.go:12]: Read that comment out loud.

File: topics/go/algorithms/sorting/selection/selection.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		for j := i; j < len(randomList); j++ {
+			if randomList[j] < randomList[i] {
+				index = j
+			}
+		}
+
+		// We will swap values based on indexes.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/sorting/selection/selection.go:19]: Don't say we, State what and more important why.

File: topics/go/algorithms/sorting/selection/selection.go (hunk 1 above)
Comment (ardan-bkennedy) [reply topics/go/algorithms/sorting/selection/selection.go:12]: Explain the what and why

Comment (ardan-bkennedy) [reply topics/go/algorithms/sorting/selection/selection.go:12]: it's not the smallest number in the list, smallest compared to randomList[i]. How can we improve the variable names for i and j?

Comment (Miraddo) [reply topics/go/algorithms/sorting/selection/selection.go:12]: Would you think this sentence could be fine and clear? 

```
// Loop to find the smallest number while compared to randomList[i], 
// If the smallest number is found, the index value changed to that index.
```
and to improve the variable names, instead, to have `i` and `j` we could write something like `leftIdx` and `smallestIdx`

File: topics/go/algorithms/sorting/selection/selection.go (hunk 2 above)
Comment (Miraddo) [reply topics/go/algorithms/sorting/selection/selection.go:19]: I got it. I should focus more on the sentences to tell the reader what is for and why we need it.
Do you think this description is clear about what and why?
```
// Swap values based on indexes, to make the list sorted
// from the smallest number to the biggest number.
```

**Comment by ardan-bkennedy** [comment]
I am reading the code and I don't really understand the what or why of it.

**Pull request merged**